docs/flow diagrams/search-api.md

```diff
@@ -0,0 +1,21 @@
+```mermaid
+sequenceDiagram
+    Client->>FhirServer: GET
+    FhirServer->>Middleware:
+    Middleware->>FhirController:
+    FhirController->>Mediatr: Search Resource
+    Mediatr->>SearchResourceHandler: SearchResourceRequest
+    SearchResourceHandler->>AuthorizationService: CheckAccess
+    AuthorizationService->>SearchResourceHandler:
+    SearchResourceHandler->>SearchService: Search by ResourceType, queries
+    SearchService->>SearchOptionsFactory: Create SearchOptions
+    SearchOptionsFactory->>SearchService:
+    SearchService->>SearchInternalAsync: Data store dependent
+    SearchInternalAsync->>SearchService: SearchResult
+    SearchService->>SearchResourceHandler: SearchResult
+    SearchResourceHandler->>Mediatr: SearchResourceResponse
```

Comment:
Would it be useful to include a step to call out that we return a bundle within the `SearchResourceResponse`?  Something like 
` SearchResourceHandler->>BundleFactory: SearchResult`
` BundleFactory->>SearchResourceHandler: Bundle`